charts/super-agent/charts/super-agent-deployment/templates/_helpers.tpl

```diff
@@ -2,19 +2,80 @@
-{{- (include "newrelic.common.naming.truncateToDNSWithSuffix" ( dict "name" "local-data" "suffix" "superagent-config" )) -}}
+{{- include "newrelic.common.naming.truncateToDNSWithSuffix" ( dict "name" "local-data" "suffix" "superagent-config" ) -}}
 {{- end -}}
 
+
+
+{{- /*
+Return the agents part that should go in the super agent config. It is created from `.Values.config.subAgents`.
+*/ -}}
+{{- define "newrelic-super-agent.config.agents.yaml" -}}
+{{- if (.Values.config).subAgents -}}
+{{- $agents := dict -}}
+{{- range $subAgentName, $subAgentConfig := (.Values.config).subAgents -}}
+  {{- if not ($subAgentConfig).type -}}
+    {{- fail (printf "Agent %s does not have agent type" $subAgentName) -}}
+  {{- end -}}
+  {{- $_ := dict $subAgentName (dict "agent_type" $subAgentConfig.type "content" $subAgentConfig.content) | mustMerge $agents -}}
+{{- end -}}
+{{- $agents | toYaml -}}
+{{- else -}}
+{{- /* Default agents for Kubernetes */ -}}
+open-telemetry:
+  type: newrelic/io.opentelemetry.collector:0.2.0
+  content:
+    chart_values:
+      global:
+        licenseKey: ${nr-env:NR_LICENSE_KEY}
+        cluster: ${nr-env:NR_CLUSTER_NAME}
+        nrStaging: ${nr-env:NR_STAGING}
+        verboseLog: ${nr-env:NR_VERBOSE}
+        region: ${nr-env:NR_REGION}
+{{- end -}}
+{{- end -}}
+
+
+
+{{- /*
+Return to which endpoint should the super agent connect to get opamp data
+*/ -}}
+{{- define "newrelic-super-agent.config.endpoints.opamp" -}}
+{{- $region := include "newrelic.common.region" . -}}
+
+{{- if eq $region "Staging" -}}
+  https://opamp.staging-service.newrelic.com/v1/opamp
+{{- else if eq $region "EU" -}}
+  https://opamp.service.eu.newrelic.com/v1/opamp
+{{- else if eq $region "US" -}}
+  https://opamp.service.newrelic.com/v1/opamp
+{{- else if eq $region "Local" -}}
+  {{- /* Accessing the value directly without protection. A developer should now how to read the error. */ -}}
+  {{ .Values.development.backend.opamp }}
+{{- else -}}
+  {{- fail "Unknown/unsupported region set for this chart" -}}
+{{- end -}}
+{{- end -}}
+
+
+
 {{- /*
 Return to which endpoint should the super agent ask to renew its token
 */ -}}
 {{- define "newrelic-super-agent.config.endpoints.tokenRenewal" -}}
-{{- if include "newrelic.common.nrStaging" . -}}
+{{- $region := include "newrelic.common.region" . -}}
+
+{{- if eq $region "Staging" -}}
   https://system-identity-oauth.staging-service.newrelic.com/oauth2/token
-{{- else if .Values.euEndpoints -}}
-  https://system-identity-oauth.service.eu.newrelic.com/oauth2/token
-{{- else -}}
+{{- else if eq $region "EU" -}}
+  https://system-identity-oauth.service.newrelic.com/oauth2/token
+{{- else if eq $region "US" -}}
   https://system-identity-oauth.service.newrelic.com/oauth2/token
+{{- else if eq $region "Local" -}}
+  {{- /* Accessing the value directly without protection. A developer should now how to read the error. */ -}}
+  {{ .Values.development.backend.tokenRenewal }}
```

Comment:
I am wondering if this is this used already or useful (have you leveraged it when creating the script)?

Reply:
I might/will be useful for development purposes for fleet team and also for us of have mock servers were we can pass tests without a valid license key/system identity.

They are going to change the API, It is a tools that does 0 cost to maintain and is useful.

It also covers all the regions that the go client supports: https://github.com/newrelic/newrelic-client-go/blob/2785c31e58443ef35d5f3872b3b930d6d595225d/pkg/region/region_constants.go#L8-L21